elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient.scala

```diff
@@ -225,16 +226,18 @@ class RestlasticSearchClient(endpointProvider: EndpointProvider, signer: Option[
 
     logger.debug(f"Got Rs request: $request (op was $op)")
 
-    val responseFuture: Future[HttpResponse] = (IO(Http) ? request).mapTo[HttpResponse]
+    val responseFuture: Future[HttpResponse] = Http().singleRequest(request)
 
-    responseFuture.map { response =>
+    responseFuture.flatMap { response =>
       logger.debug(f"Got Es response: $response")
+      val entityFuture = AkkaHttpUtil.entityToString(response.entity)
       if (response.status.isFailure) {
-        logger.warn(s"Failure response: ${response.entity.asString.take(500)}")
+        val entityString = Await.result(entityFuture, 100.millis)
```

Comment:
Maybe cleaner to use `.recover` or `.recoverWith` to handle the failure without having to Await on the entity?